src/carousel.tsx

```diff
@@ -457,6 +464,24 @@ const BaseCarousel: React.FC<BaseCarouselProps> = ({
 
     const shouldAutoplay = useShouldAutoplay(!!autoplay, carouselRef);
 
+    React.useEffect(() => {
+        if (
+            initialActiveItem &&
+            (initialActiveItem !== initialItemInfo.current.initialItem ||
+                itemsPerPageFloor !== initialItemInfo.current.itemsPerPage)
+        ) {
+            goToPage(Math.floor(initialActiveItem / itemsPerPageFloor));
+            initialItemInfo.current.initialItem = initialActiveItem;
+            initialItemInfo.current.itemsPerPage = itemsPerPageFloor;
+        }
+    }, [
+        initialActiveItem,
+        itemsPerPageFloor,
+        goToPage,
+        initialItemInfo.current.initialItem,
+        initialItemInfo.current.initialItem,
+    ]);
```

Comment:
```suggestion
    }, [initialActiveItem, itemsPerPageFloor, goToPage]);
```
You don't need refs in `useEffect` dependencies